core/controllers/acl_decorators.py

```diff
@@ -846,7 +846,8 @@ def test_can_rate(self, exploration_id, **kwargs):
             UnauthorizedUserException: The user does not have credentials to
                 rate an exploration.
         """
-        if (role_services.ACTION_RATE_ANY_PUBLIC_EXPLORATION in
+        if (
```

Comment:
I strongly encourage for the second part -- write an email/doc showing examples and confirm that your understanding is right. This will save ample amount of time IMO.

Reply:
I did this because in the past I've asked many times to break after the `if (` by @DubeySandeep so I was thinking that we also want this check for `if` and `elif` as well, but I think you are right I will exclude these `if`, `else` and `while` from this check.

Reply:
Reverted the changes as well.